beacon-chain/state/getters.go

```diff
@@ -1016,6 +1017,44 @@ func (b *BeaconState) currentJustifiedCheckpoint() *ethpb.Checkpoint {
 	return b.safeCopyCheckpoint(b.state.CurrentJustifiedCheckpoint)
 }
 
+// MatchCurrentJustifiedCheckpoint returns true if input justified checkpoint matches
+// the current justified checkpoint in state.
+func (b *BeaconState) MatchCurrentJustifiedCheckpoint(c *ethpb.Checkpoint) bool {
+	if !b.HasInnerState() {
+		return false
+	}
+	if b.state.CurrentJustifiedCheckpoint == nil {
+		return false
+	}
+
+	b.lock.RLock()
```

Comment:
Why do we need locks here? We are just accessing struct fields for reading, there shouldn't be any contention across goroutines

Reply:
Ah i see. Removing...